kernel/src/table_changes/mod.rs

```diff
@@ -195,6 +187,7 @@ impl TableChanges {
             log_segment,
             start_version,
             schema,
+            start_table_config: start_snapshot.table_configuration().clone(),
```

Comment:
can we avoid the clone? aren't we throwing away the start_snapshot? oh i guess if this returns ref we need to clone.. hm maybe there's a case for some into_config conversions etc.? (thinking out loud - not to solve here)

Reply:
Yeah start_snapshot is Arc, so we probably can't take.